autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java

```diff
@@ -194,16 +198,17 @@ private void scaleDownByNodeIds(List<String> decommissionNodeIds) {
                     "decommissionNodeIds '{}', error '{}' ", hostGroup, cluster.getStackCrn(), decommissionNodeIds.size(),
                     decommissionNodeIds, statusReason, e);
         } finally {
-            createHistoryAndNotify(-decommissionNodeIds.size(), totalNodes, statusReason, scalingStatus);
+            processAutoscalingTriggered(-decommissionNodeIds.size(), totalNodes, statusReason, scalingStatus);
         }
     }
 
-    private void createHistoryAndNotify(int adjustmentCount, int totalNodes, String statusReason, ScalingStatus scalingStatus) {
+    private void processAutoscalingTriggered(int adjustmentCount, int totalNodes, String statusReason, ScalingStatus scalingStatus) {
         History history = historyService.createEntry(scalingStatus,
                 StringUtils.left(statusReason, STATUSREASON_MAX_LENGTH), totalNodes, adjustmentCount, policy);
         notificationSender.sendHistoryUpdateNotification(history, cluster);
         auditService.auditAutoscaleServiceEvent(scalingStatus, statusReason, cluster.getStackCrn(),
                 cluster.getClusterPertain().getTenant(), System.currentTimeMillis());
+        usageReportingService.reportAutoscalingTriggered(adjustmentCount, totalNodes, scalingStatus, statusReason, policy.getAlert(), cluster);
```

Comment:
If scaling operation is already in progress LoadBased autoscaling wont trigger another event so there would not be that metric here. If time-based scaling is configured by user such that every 2 mins another scaling event is triggered this metric will help us to fine tune the validation required for schedule-based autoscaling.
Other flows generate similar events for all CB Flow changes and the number of events generated here will be far less. 
I have tested multiple times and there is no difference between actual scaling node count and yarn recommended node count. This information we already get from max node count in LoadAlertDefinition.

Reply:
The difference would come in when the YARN recommended node count is much higher than the maxNodeCount configured for the cluster?
On the downscale, the difference will be - YARN recommends N nodes, we need to decommission >N nodes - so which ones/how many were a result of YARN, and which ones/how many were randomly picked after this.

Reply:
The forced nodes autoscale downsized in case of LoadAlertConfig limits vs the ones recommended by YARN are all already available in Kibana. I dont agree such detailed information ( node-ids, ips, routine operation logs)  are all should be part of telemetry metrics. This will add volume to telemetry storage also.

Reply:
At the same time, it is the best mechanism we have at the moment to correlate scaling operations/failures with what YARN had to say. Ideally, would like the YARN API to return metrics it used to come to the decision, which could also be part of a string dump (and a non queryable field in the final data consumption path).

Can add that as a follow up - I think this can be quite useful though, especially if YARN can give back the metrics that it used for the specific decision.

Reply:
ok.